dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java

```diff
@@ -0,0 +1,164 @@
+            //The size of the detail should have grown as we insert new triggers.
+            assertEquals(detail.size(), i);
+        }
+
+        //Verify at least 1 job has been launched.
+        final Optional<JobExecutionContext> jobExecutionContext = getJobExecutionContext();
+        assertTrue(jobExecutionContext.isPresent());
+
+        //Now lets wait for the all the threads to complete.
+        MyStatefulJob.countDownLatch.await();
+
+        //Now lets revise the execution times. Since they were supposed to run sequentially the should never overlap.
+        final HashSet<MyStatefulJob> myStatefulJobs = new HashSet<>(MyStatefulJob.finishedJobs);
+        final Iterator<MyStatefulJob> iterator = myStatefulJobs.iterator();
+        while(iterator.hasNext()){
+            final MyStatefulJob myStatefulJob = iterator.next();
+            iterator.forEachRemaining(job -> {
+                //They should never overlap.
+                assertFalse(myStatefulJob.getTimeRange().overlaps(job.getTimeRange()));
+            });
+        }
+    }
+
+    /**
+     * This gets you the execution context for the given job-name.
+     * @return
+     */
+    private Optional<JobExecutionContext> getJobExecutionContext(){
+        final String jobName = DotStatefulJob.getJobName(MyStatefulJob.class);
+        try {
+            final Scheduler sequentialScheduler = getSequentialScheduler();
+            @SuppressWarnings("unchecked")
+            final List<JobExecutionContext> executingJobs = sequentialScheduler.getCurrentlyExecutingJobs();
+            return executingJobs.stream().filter(jobExecutionContext -> {
+                final JobDetail jobDetail = jobExecutionContext.getJobDetail();
+                return jobDetail != null && jobName.equals(jobDetail.getName());
+            }).findFirst();
+        } catch (Exception e) {
+            Logger.error(DotStatefulJobTest.class, "Error retrieving execution context. " , e);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * additional struct to follow execution time lapse.
+     */
+    public static class LocalTimeRange {
+
+        private final LocalTime from;
+        private final LocalTime to;
+
+        LocalTimeRange(final LocalTime from, final LocalTime to) {
+            requireNonNull(from, "from must not be null");
+            requireNonNull(to, "to must not be null");
+            this.from = from;
+            this.to = to;
+        }
+
+        boolean overlaps(final LocalTimeRange other) {
+            requireNonNull(other, "other must not be null");
+            return isBetween(other.from, this.from, this.to)
+                    || isBetween(other.to, this.from, this.to)
+                    || isBetween(this.from, other.from, other.to)
+                    || isBetween(this.to, other.from, other.to);
+        }
+
+        private static boolean isBetween(final LocalTime time, final LocalTime from, final LocalTime to) {
```

Comment:
Codacy found an issue: [Avoid variables with short names like to](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077927)